provisioning/transport/http/src/ProvisioningTransportHandlerHttp.cs

```diff
@@ -174,11 +207,12 @@ public ProvisioningTransportHandlerHttp()
             catch (HttpOperationException oe)
```

Comment:
any value in having this catch block: https://github.com/Azure/azure-iot-sdk-csharp/blob/bikamani/retryAfter/provisioning/transport/http/src/ProvisioningTransportHandlerHttp.cs#L207-L235?

Reply:
yes for RegisterDeviceAsync at line 117

Reply:
it does nothing more than throw a `ProvisioningTransportException`. You can merge both the catch blocks into a single block: if exception thrown is not PTE, then evaluate if it is transient, and throw the corresponding PTE.

Reply:
Can there not be other type of exception than HttpOperationException?

Reply:
the register operation will only throw an HttpOperationException, that's what the impl does afaik. But your catch block would handle other cases as well.

```
catch (Exception ex) when (!(ex is ProvisioningTransportException))
{
	if (Logging.IsEnabled) Logging.Error(
		this,
		$"{nameof(ProvisioningTransportHandlerHttp)} threw exception {ex}",
		nameof(RegisterAsync));

	bool isTransient = false;
	if (ex is HttpOperationException)
	{
		HttpOperationException oe = (HttpOperationException)ex;
		isTransient = oe.Response.StatusCode >= HttpStatusCode.InternalServerError || HttpStatusCode.TooManyRequests == oe.Response.StatusCode;

		try
		{
			var errorDetails = JsonConvert.DeserializeObject<ProvisioningErrorDetailsHttp>(oe.Response.Content);
			throw new ProvisioningTransportException(oe.Response.Content, oe, isTransient, errorDetails);
		}
		catch (JsonException jex)
		{
			if (Logging.IsEnabled) Logging.Error(
				this,
				$"{nameof(ProvisioningTransportHandlerHttp)} server returned malformed error response." +
				$"Parsing error: {jex}. Server response: {oe.Response.Content}",
				nameof(RegisterAsync));

			throw new ProvisioningTransportException(
				$"HTTP transport exception: malformed server error message: '{oe.Response.Content}'",
				jex,
				false);
		}
	}

	throw new ProvisioningTransportException($"HTTP transport exception", ex, isTransient);
}
```

Reply:
now that i've written this, I don't have a very strong preference to change it; I prefer it this way, but you can leave as is, if you like.

Reply:
yeah agree, that's more of a preference and how you like to see. I find less nesting is more readable